config-store-yaml/src/main/java/io/stargate/config/store/api/yaml/ConfigStoreYaml.java

```diff
@@ -18,22 +18,36 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.type.MapType;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Ticker;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
 import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.UncheckedExecutionException;
 import io.stargate.config.store.api.ConfigStore;
 import io.stargate.config.store.api.ConfigWithOverrides;
 import io.stargate.config.store.api.MissingModuleSettingsException;
-import java.io.IOException;
-import java.io.UncheckedIOException;
 import java.nio.file.Path;
+import java.time.Duration;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import javax.annotation.Nonnull;
 
 public class ConfigStoreYaml implements ConfigStore {
   private final ObjectMapper mapper;
   private final Path configFilePath;
   private final MapType yamlConfigType;
+  @VisibleForTesting final LoadingCache<Path, Map<String, Map<String, Object>>> configFileCache;
+  public static final Duration DEFAULT_EVICTION_TIME = Duration.ofSeconds(30);
```

Comment:
nit: Would it be worth adding a java system property for this?

Reply:
I am not sure if we want to make it configurable. This is a very internal setting. I thought that maybe we could expose `stats()` of this cache via MericRegistry: https://github.com/stargate/stargate/pull/343/files#diff-249fc05c8cbd5acea87e6ffab58018865968927f4f6741745c47b3af3a5efcf6R65
observe it on production and adapt this parameter accordingly (if needed).

Reply:
sgtm :+1: